### PR TITLE
Adopt ApprovalType from core

### PR DESCRIPTION
**Pull request by vinistevam**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

Adopt `ApprovalType` to approval requests/confirmations leaving room for a more cohesive usage between extension and mobile.
_1. Align approval requests and confirmations between mobile and extension.

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

resolves: https://github.com/MetaMask/mobile-planning/issues/795

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

File: app/components/Nav/Main/RootRPCMethodsUI.js [hunk 1]
```diff
@@ -399,9 +398,14 @@ const RootRPCMethodsUI = (props) => {
   const toggleExpandedMessage = () =>
     setShowExpandedMessage(!showExpandedMessage);
 
+  const isSigningApprovalTypeValid = (type) =>
```
Comment (matthewwalsh0) [inline app/components/Nav/Main/RootRPCMethodsUI.js:401]: `isSigningApprovalType`?

File: app/components/Nav/Main/RootRPCMethodsUI.js [hunk 2]
```diff
@@ -399,9 +398,14 @@ const RootRPCMethodsUI = (props) => {
...
+    type === ApprovalType.signType ||
```
Comment (matthewwalsh0) [inline app/components/Nav/Main/RootRPCMethodsUI.js:402]: Shouldn't this be `ApprovalType.PersonalSign`?

File: docs/confirmations.md [hunk 3]
```diff
@@ -72,14 +72,14 @@ await Engine.context.ApprovalController.updateRequestState({
 
 ### 3. Define Approval Type
 
-Update the `ApprovalTypes` enum in [app/core/RPCMethods/RPCMethodMiddleware.ts](../app/core/RPCMethods/RPCMethodMiddleware.ts) to include a new entry to identify the new confirmations.
+Update the `ApprovalType` enum in [@metamask/controller-utils](https://github.com/MetaMask/core/blob/main/packages/controller-utils/src/constants.ts) to include a new entry to identify the new confirmations.
```
Comment (matthewwalsh0) [inline docs/confirmations.md:75]: ❤️

File: app/core/RPCMethods/RPCMethodMiddleware.ts [hunk 4]
```diff
@@ -285,7 +280,7 @@ export const getRpcMethodMiddleware = ({
             if (!approved) {
               // Prompts user approval UI in RootRPCMethodsUI.js.
               await requestUserApproval({
-                type: ApprovalTypes.CONNECT_ACCOUNTS,
+                type: ApprovalType.ConnectAccounts,
```
Comment (matthewwalsh0) [inline app/core/RPCMethods/RPCMethodMiddleware.ts:283]: I think there's a few more like this in `wallet_addEthereumChain` and `wallet_switchEthereumChain` if you search for `requestUserApproval`.

Comment (vinistevam) [reply app/core/RPCMethods/RPCMethodMiddleware.ts:283]: I found 4 extra places and applied the change to use the `ApprovalType` from core.

File: package.json [hunk 5]
```diff
@@ -148,7 +148,7 @@
     "@metamask/base-controller": "^1.1.2",
     "@metamask/composable-controller": "^1.0.2",
     "@metamask/contract-metadata": "^2.1.0",
-    "@metamask/controller-utils": "^2.0.0",
+    "@metamask/controller-utils": "^3.4.0",
```
Comment (matthewwalsh0) [inline package.json:151]: Is this safe to do? What was the breaking change?

**Review by matthewwalsh0 — APPROVED** [review]

File: package.json (hunk 5 above)
Comment (vinistevam) [reply package.json:151]: The breaking change was to remove `isomorphic-fetch` but there is an issue with `NetworksChainId` that might be related to assets-controller so better to update to the v40-42 core releases. Meanwhile, I will try to update the packages in this PR and test if I can fix this issue.

**Review by bergarces — APPROVED** [review]

File: app/core/RPCMethods/RPCMethodMiddleware.ts
Comment (bergarces) [inline app/core/RPCMethods/RPCMethodMiddleware.ts:1]: On line 195, the method `requestUserApproval` is currently expecting an object with a `type: string`.

We might want to narrow down that type to `ApprovalType`.

**Pull request closed without merging**